lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -232,6 +232,42 @@ findModule(ASTContext &ctx, AccessPathElem moduleID,
                           moduleBuffer, moduleDocBuffer, scratch);
 }
 
+static std::pair<StringRef, clang::VersionTuple>
+getOSAndVersionForDiagnostics(const llvm::Triple &triple) {
```

Comment:
I don't love adding *another* one of these, but neither `PlatformKind` nor `DarwinPlatformKind` has all the features we need here, and I can't figure out a good way to unify them.